acceptance/Rakefile

```diff
@@ -1,30 +1,357 @@
 
-    task :aio do
-      if (ENV['SHA']) && (not ENV['SUITE_COMMIT'])
-        ENV['SUITE_COMMIT'] = ENV['SHA']
-      end
-      unless ENV['SUITE_COMMIT'] then
-        fail('SUITE_COMMIT or SHA environment variable must be set to the SHA of the puppet-agent package to test')
+module HarnessOptions
+
+  DEFAULTS = {
+    :type => 'aio',
+    :helper => ['lib/helper.rb'],
+    :tests  => ['tests'],
+    :log_level => 'info',
+    :color => false,
+    :root_keys => true,
+    :ssh => {
+      :keys => ["~/.ssh/id_rsa-acceptance"],
+    },
+    :xml => true,
+    :timesync => false,
+    :repo_proxy => true,
+    :add_el_extras => true,
+    :preserve_hosts => 'onfail',
+  }
+
+  class Aggregator
+    attr_reader :mode
+
+    def initialize(mode)
+      @mode = mode
+    end
+
+    def get_options(file_path)
+      puts file_path
+      if File.exists? file_path
+        options = eval(File.read(file_path), binding)
+      else
+        puts "No options file found at #{File.expand_path(file_path)}"
       end
+      options || {}
+    end
+
+    def get_mode_options
+      get_options("./config/#{mode}/options.rb")
+    end
+
+    def get_local_options
+      get_options("./local_options.rb")
+    end
+
+    def final_options(intermediary_options = {})
+      mode_options = get_mode_options
+      local_overrides = get_local_options
+      final_options = DEFAULTS.merge(mode_options)
+      final_options.merge!(intermediary_options)
+      final_options.merge!(local_overrides)
+      return final_options
+    end
+  end
+
+  def self.options(mode, options)
+    final_options = Aggregator.new(mode).final_options(options)
+    final_options
+  end
+end
+
+def beaker_test(mode = :packages, options = {})
+  final_options = HarnessOptions.options(mode, options)
+
+  if mode == :git
+    final_options[:install] ||= []
+    final_options[:install] << "#{build_giturl('pxp-agent')}##{sha}"
+  end
+
+  options_file = 'merged_options.rb'
+  File.open(options_file, 'w') do |merged|
+    merged.puts <<-EOS
+# Copy this file to local_options.rb and adjust as needed if you wish to run
+# with some local overrides.
+EOS
+    merged.puts(final_options.pretty_inspect)
+  end
+
+  tests = ENV['TESTS'] || ENV['TEST']
+  tests_opt = "--tests=#{tests}" if tests
 
-      if not ENV['TEST_TARGET']
-        fail "TEST_TARGET environment variable must be set to the name of a host config e.g. redhat-7-x86_64"
+  config_opt = "--hosts=#{config}" if config
+
+  overriding_options = ENV['OPTIONS']
+
+  args = ["--options-file", options_file, config_opt, tests_opt, overriding_options].compact
+
+  begin
+    sh("beaker", *args)
+  ensure
+    if (hosts_file = config || final_options[:hosts_file]) && hosts_file !~ /preserved_config/
+      cp(hosts_file, "log/latest/config.yml")
+      generate_config_for_latest_hosts if final_options[:preserve_hosts] || overriding_options =~ /--preserve-hosts/
+    end
+    mv(options_file, "log/latest")
+    puts "\n\n"
```

Comment:
most of this ensure is no longer needed as beaker saves its preserved_hosts file and config in one place.